src/arch/riscv/traps.S

```diff
@@ -115,14 +115,17 @@ handle_syscall:
 
 #ifdef CONFIG_FASTPATH
   li t3, SYSCALL_CALL
-  beq a7, t3, c_handle_fastpath_call
-
+  bne a7, a3, 1f
+  j c_handle_fastpath_call
+1:
   li t3, SYSCALL_REPLY_RECV
 #ifdef CONFIG_KERNEL_MCS
   /* move reply to 3rd argument */
   mv a2, a6
 #endif
-  beq a7, t3, c_handle_fastpath_reply_recv
+  bne a7, a3, 1f
+  j c_handle_fastpath_reply_recv
+1:
```

Comment:
I would give this label a different name, it weird to have two labels with the same name in the same function.